assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties

```diff
@@ -58,7 +58,7 @@ che.workspace.auto_start=true
 # operations that require asynchronous execution e.g. starting/stopping
 
 # possible values are 'fixed', 'cached'
-che.workspace.pool.type=fixed
+che.workspace.pool.type=cached
```

Comment:
Previously, we had cached thread pool for JSON RPC messages and if I'm not mistaken it caused OutOfMemory crashing...
If we set cached without any limit - I assume it can lead to OutOfMemory crashing as well.
With JSON RPC it was easier to reproduce, but still...

Maybe it makes sense to consider this value to be configured along with the memory that Che container gets.
Like in `che.properties` it's fixed and every deployment provides his own values according to provided container memory.
To be more safe we could instead of reusing default java cached pool, introduce our own cached pool with a maximum size. We could estimate some maximum size according to container memory.

I'm not a big expert in this area, and it's just thoughts. In reality, my comments may be not so critical or not relevant at all.

Reply:
yeah, this sounds relevant - having no ability of spesifying `maximum size` is potentially risky. but again, according to the docs: 

> Threads that have not been used for sixty seconds are terminated and removed from the cache. Thus, a pool that remains idle for long enough will not consume any resources

So, it looks like technically we should not care about  `maximum size` (if we should than there is an issue on the other level e.g. RAM misconfig etc)